### PR TITLE
Fix TautomerTransform in operator=

### PR DESCRIPTION
**Pull request by bp-kelley**
Fixes a serious bug (operator= doesn't return a value) and suppresses another warning.

### CODE REVIEW

File: Code/GraphMol/FileParsers/MolWriters.h [hunk 1]
```diff
@@ -210,7 +210,7 @@ class RDKIT_FILEPARSERS_EXPORT TDTWriter : public MolWriter {
   TDTWriter(const std::string &fileName);
   TDTWriter(std::ostream *outStream, bool takeOwnership = false);
 
-  ~TDTWriter();
+  ~TDTWriter() throw();
```
Comment (greglandrum) [inline Code/GraphMol/FileParsers/MolWriters.h]: It's worth thinking about being systematic about including `throw()` specs, but I think it should be done in a separate PR.
Given that the dtor here ends up calling other methods that are not marked `throw()` this is almost certainly wrong anyway.

Comment (bp-kelley) [reply Code/GraphMol/FileParsers/MolWriters.h]: To be clear, this means remove throw from both interface and implementation?

Comment (bp-kelley) [reply Code/GraphMol/FileParsers/MolWriters.h]: Ok, I’ll do a separate pr.
